### PR TITLE
Nested relative local path bower components cannot resolve path #1513

### PR DESCRIPTION
**Pull request by federicobarera**
Implementation follows dependencies' cwd,  if specified.

### CODE REVIEW

File: lib/core/Manager.js [hunk 1]
```diff
@@ -481,6 +482,23 @@ Manager.prototype._parseDependencies = function (decEndpoint, pkgMeta, jsonKey)
         childDecEndpoint.unresolvable = !!decEndpoint.unresolvable;
 
         // Otherwise, just fetch it from the repository
+        
+        //Adjust dependency source if relative, considering its own configuration, if any
+        if (/^(\.|\.\.)\//g.test(childDecEndpoint.source)) {
```
Comment (samccone) [inline lib/core/Manager.js]: I do not think this will work on win32 systems due to the `/` and relative path correct?

**Comment by samccone** [comment]
@federicobarera I am not sure I totally understand the motivation behind this one, care to explain a bit more?

**Comment by federicobarera** [comment]
I saw a pull request regarding the same bug, done in the resolverFactory. That was resolving the path of the relative dependencies correctly, but, if the nested dependency had a .bowerrc file with a different cwd, the resolution was failing. This is the motivation behind this pull request. 

If the expected behavior is not to follow the configuration of dependencies while resolving, this can be ignored. 

I haven't verified the behavior on win32, and probably extend the regex to \.

**Comment by federicobarera** [comment]
backslash

**Comment by federicobarera** [comment]
Sorry, let me explain a bit better:

You have this regex in the resolverFactory, which can be used instead to detect relative paths.

`^\.\.?[\/\\]`, which solve the win32 issue.

The point now is that if you want to consider sub dependencies configurations while resolving relative dependencies...
From what I can see, please correct me if I am wrong, when the install command runs, there is one project object, with one configuration and one manager object which take care of resolving everything. 
The same configuration is used to resolve the paths of the dependencies, which is wrong.

In this pull request: https://github.com/bower/bower/pull/1737 
he corrected the path of the dependency, without considering the dependency configuration, which I believe is right if you consider that you want just one main project with one configuration. The pull request is still pending tho.

In my version I also check if the dependency has its own configuration, where for example, you can set another cwd for this project. That cwd should be used to resolve the dependency of my project, even if this project is a dependency.

This is the assumption behind this implementation.
I agree that a cleaner code in this scenario, would recursively create sub "Project" objects with new configurations and resolvers, but this would largely impact the current structure.

Please let me know if my assumption is incorrect and/or if you guys have plans to fix this bug.

Thanks

**Comment by federicobarera** [comment]
Any news in matter?

**Comment by faceleg** [comment]
Sorry @federicobarera I'm not sure we fully understand what this PR is trying to solve?

**Comment by federicobarera** [comment]
The problem is with local bower packages references.
Assume you have bower project A defining a local reference to bower proj B like:

"ProjectB" : "../../ProjectB/"

If Project B internally references project C like

"ProjectC : "./ProjectC/" 

currently bower will fail the resolution as it will start resolve the dependency from ProjectA folder.

My PR starts the dep resolutions from the dependency itself. Moreover I try to check if the dependency defines a local cwd, in that case I start from there

**Comment by faceleg** [comment]
That makes more sense (never used bower that way).

@bower/bower-volunteers independent review please

**Comment by faceleg** [comment]
Wow first time I've ever seen appveyor actually *like* a PR!

**Comment by federicobarera** [comment]
Thanks, please let me know how it goes / if you guy intend to support this usage of bower.
We (old company) had our frontend framework divided in tens of bower modules saved in different directories of the same repository.
Each compiled version of the framework used different combination of packages, but with the current build of bower was not possible to resolve the path of those correctly, forcing us to create this custom version of bower.

I'll update the pull request with the updated regex spotted by sam.

**Comment by sheerun** [comment]
This PR is very old and conflicting plus Bower is depreacted and we recommend switching to Yarn

**Pull request closed without merging**